google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -200,6 +200,19 @@ public void serverTimestamp() throws Exception {
     assertTrue(documentSnapshot.getDate("time").getTime() > 0);
   }
 
+  @Test
+  public void updateMicrosecondTimestamp() throws Exception {
+    DocumentReference documentReference = addDocument("time", Timestamp.ofTimeSecondsAndNanos(0, 123000));
+    DocumentSnapshot documentSnapshot = documentReference.get().get();
+
+    Timestamp timestamp = documentSnapshot.getTimestamp("time");
+    documentReference.update("time", timestamp);
```

Comment:
I might be missing something, but shouldn't this update the timestamp to a different value than the original?

Reply:
I renamed the test to `timestamptDoesntGetTruncatedDuringUpdate`.  This test doesn't use server timestamps anymore, since those only have millisecond precision. Instead, the same value is written, read, and re-written.